### PR TITLE
Remove nemsio dependency

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

Remove nemsio from the FV3 write component. nemsio is legacy file format that is being deprecated. Since there are no regression tests using nemsio file format, no change of answers is expected.

### Issue(s) addressed

https://github.com/ufs-community/ufs-weather-model/issues/207

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] CI https://github.com/ufs-community/ufs-weather-model/actions/runs/1060865339

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?

- waiting on noaa-emc/fv3atm/pull/339

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.

File: CMakeLists.txt [hunk 1]
```diff
@@ -207,7 +202,7 @@ endif()
 ### Data Components [NEMSdatm, CDEPS]
 ###############################################################################
 if(NEMSdatm)
-  add_subdirectory(DATM)
+  add_subdirectory(DATM/DATM)
```
Comment (junwang-noaa) [inline CMakeLists.txt:205]: Why do we need this change for removing nemsio?

**Review by climbfuji — APPROVED** [review]
Approved, but I am also interested in the answer to Jun's question (DATM -> DATM/DATM).

File: CMakeLists.txt (hunk 1 above)
Comment (DusanJovic-NOAA) [reply CMakeLists.txt:205]: `add_subdirectory(DATM)` adds both `DATM` and `tools` which depends on nemsio. ufs-weather-model depends only on `DATM/DATM`

Comment (aerorahul) [reply CMakeLists.txt:205]: I agree with this change, unless the `tools` are pulled out, which they should be.  The `tools` should ideally reside in `UFS_utils`

Comment (aerorahul) [reply CMakeLists.txt:205]: Or `NEMS-datm` or something like that.

Comment (DeniseWorthen) [reply CMakeLists.txt:205]: I don't believe that what is currently in the tools subdirectory is used at all. It is the original code I wrote to convert nesmio to netcdf for the DATM. 

The code which is currently used to generate the DATM files resides in the utilities subdirectory of the [DATM-MOM6-CICE6](https://github.com/NOAA-EMC/DATM-MOM6-CICE5/tree/develop/utilities) repo. That also contains the utility to create the SCRIP file for the DATM (gen_scrip) although I'm not 100% sure it is used for CDEPS. It is the utilities subdirectory which would need to get moved to ufs-utils.

Comment (junwang-noaa) [reply CMakeLists.txt:205]: @binli2337 @DeniseWorthen can we remove tools directory under https://github.com/noaa-emc/nemsdatm to avoid confusion?

Comment (DeniseWorthen) [reply CMakeLists.txt:205]: Yes, I believe we can & should remove the tools subdirectory.

Comment (binli2337) [reply CMakeLists.txt:205]: Yes.

**Comment by BrianCurtis-NOAA** [comment]
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/686959771/20210723191511/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_72855
Test control_2threads 018 failed failed 
Test control_2threads 018 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Review by DeniseWorthen — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**